--- libstdc++-v3/libsupc++/eh_ptr.cc	2012-02-11 03:20:43.000000000 +0900
+++ libstdc++-v3/libsupc++/eh_ptr.cc	2014-10-15 17:51:55.362084375 +0900
@@ -217,6 +217,9 @@
   __GXX_INIT_DEPENDENT_EXCEPTION_CLASS(dep->unwindHeader.exception_class);
   dep->unwindHeader.exception_cleanup = __gxx_dependent_exception_cleanup;
 
+  __cxa_eh_globals *globals = __cxa_get_globals ();
+  globals->uncaughtExceptions += 1;
+
 #ifdef _GLIBCXX_SJLJ_EXCEPTIONS
   _Unwind_SjLj_RaiseException (&dep->unwindHeader);
 #else
